build_defs/antlr.bzl

```diff
@@ -0,0 +1,54 @@
+#-----------------------------------------------------------------------------
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#-----------------------------------------------------------------------------
+"""
+Generates C++ parser, visitor, listener, lexer from a grammar file.
+"""
+
+load("@rules_antlr//antlr:antlr4.bzl", "antlr")
+
+def antlr_cc_library(name, grammar_file, package = None, listener = True, visitor = True):
```

Comment:
It is not like we are going to generate multiple parsers.  Do we even need to define this bzl rule? Can't we directly have the `antlr` and `cc_library` targets in a BUILD file?

Reply:
We can have them in the BUILD file. Makes my life much easier. I am going to make this change.